### ISSUE TITLE
Split foz media files from the shaders pre-cache

### ISSUE THREAD

**Issue by Galcian79** [ValveSoftware/steam-for-linux#9132]
Since radv would get support for the graphics pipeline libraries very soon, so the Steam Deck, and i guess it will be the preferred path, it would be nice to be able to to turn off the Steam pre-cache feature, saving both disk space and bandwidth, without affecting the transcoded media folder. So i would ask to introduce separate settings for those two features.

**Comment by cchulo** [comment]
to add on to this, have a way to manually redownload these videos, or restore it whenever verify integrity of game files is executed

I just reinstalled my OS, so my ~/.steam folder was not backed up due to just mindlessly assuming shaders are wasted space in the backup, but forgot that games like RE:HD have transcoded videos that are required for the cutscenes. I tried everything to avoid having to redownload the game, but alas, the only solution is to redownload the game again in order for the transcoded video to be redownloaded

**Comment by polluxau** [comment]
I'd also like to see this be improved, too many new Linux users keep asking why shader cache is a thing or if it's ok to turn off 🫠

**Comment by Reonu** [comment]
To add to this, it would be good if we were able to toggle the shader pre-caching feature on a per-game basis. Similarly to the background downloads setting, there could be a global setting (on/off) and a per-game override (use global setting/on/off).

Some games take ages to compile shaders, others don't. It's a big help in some games, and useless in others. Considering the amount of background CPU work it adds, or the delay in opening games if you disable that, it shouldn't just be a global setting.

**Comment by Hasshu** [comment]
The fact that one has to scour GitHub or Reddit to learn this feature exists at all... is bizarre. Curiously enough, there's at least one "Deck Verified" game – *Darksiders Genesis* – which won't play videos unless you enable shader pre-caching.

Make of that what you will.

**Comment by cchulo** [comment]
> The fact that one has to scour GitHub or Reddit to learn this feature exists at all... is bizarre. Curiously enough, there's at least one "Deck Verified" game – *Darksiders Genesis* – which won't play videos unless you enable shader pre-caching.
> 
> Make of that what you will.

I have gotten around this with proton-ge, have you tried this at all? the main challenge is the fact that valve cannot distribute certain codecs due to licensing. I think proton-ge downloads them to a game's prefix during initial load. You could also try using protontricks to add codecs. Hoops to jump through, but this is the state of linux gaming as of now. The bright side is that this issue mostly comes up with older games that use CGI cutscenes, not so much with modern games that feature in-engine cutscenes.

**Comment by polluxau** [comment]
> The fact that one has to scour GitHub or Reddit to learn this feature exists at all... is bizarre. Curiously enough, there's at least one "Deck Verified" game – *Darksiders Genesis* – which won't play videos unless you enable shader pre-caching.
> 
> Make of that what you will.

Question do you have h.264 enabled in steam? This can be disabled by default when you install steam sometimes, you can check by trying to watch a broadcast in steam, if the broadcast comes up with a error it's most likely it's disabled 

You can enable h.264 in steam by doing 

steam steam://unlockh264/

In terminal then test the game that doesn't work with the codecs again

**Comment by Hasshu** [comment]
> proton-ge
> valve cannot distribute certain codecs due to licensing
> protontricks

@cchulo Yep, am aware of those. I just felt that calling *Genesis* "verified" is a bit dishonest, especially considering that the earlier *Darksiders II* and *III* have working cutscenes out of the box. Apparently, I was wrong (see below).

> steam steam://unlockh264/

@polluxau Now, I did know that broadcasting on Linux may require this kind of tinkering, but had no idea the codec in question is used by Proton as well. A-a-and... *Genesis* seems to have working cutscenes now. Thanks a lot!

Yet another thing that should be made more obvious to users, methinks.

\</off-topic>

**Comment by polluxau** [comment]
> > proton-ge
> > valve cannot distribute certain codecs due to licensing
> > protontricks
> 
> @cchulo Yep, am aware of those. I just felt that calling *Genesis* "verified" is a bit dishonest, especially considering that the earlier *Darksiders II* and *III* have working cutscenes out of the box. Apparently, I was wrong (see below).
> 
> > steam steam://unlockh264/
> 
> @polluxau Now, I did know that broadcasting on Linux may require this kind of tinkering, but had no idea the codec in question is used by Proton as well. A-a-and... *Genesis* seems to have working cutscenes now. Thanks a lot!
> 
> Yet another thing that should be made more obvious to users, methinks.
> 
> \</off-topic>

Yeah I had a similar problem in another game and a valve employee told me to try that and it worked instantly, don't know why it's disabled by default
